saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/OpenSamlAuthenticationRequestFactory.java

```diff
@@ -67,4 +70,18 @@ public void setClock(Clock clock) {
 		Assert.notNull(clock, "clock cannot be null");
 		this.clock = clock;
 	}
+
+	/**
+	 * Sets the {@code protocolBinding} to use when generating authentication requests
+	 *
```

Comment:
Can you add to this documentation
1. What values are acceptable
2. @throws for when invalid parameter is set

I wonder if we should switch to `IllegalArgumentException` since this is not really SAML, but just a simple getter that takes a value.